docs/data/material/components/toggle-button/CustomizedDividers.tsx

```diff
@@ -18,15 +18,18 @@ const StyledToggleButtonGroup = styled(ToggleButtonGroup)(({ theme }) => ({
   '& .MuiToggleButtonGroup-grouped': {
     margin: theme.spacing(0.5),
     border: 0,
+    borderRadius: theme.shape.borderRadius,
     '&.Mui-disabled': {
       border: 0,
     },
-    '&:not(:first-of-type)': {
-      borderRadius: theme.shape.borderRadius,
-    },
-    '&:first-of-type': {
-      borderRadius: theme.shape.borderRadius,
-    },
+  },
+  '& .MuiToggleButtonGroup-middleButton': {
+    marginLeft: -1,
+    borderLeft: '1px solid transparent',
+  },
+  '& .MuiToggleButtonGroup-lastButton': {
+    marginLeft: -1,
+    borderLeft: '1px solid transparent',
```

Comment:
We can combine this:
```suggestion
  '& .MuiToggleButtonGroup-middleButton,& .MuiToggleButtonGroup-lastButton': {
    marginLeft: -1,
    borderLeft: '1px solid transparent',
  },
```
Also, now, is this considered a breaking change for developers using those pseudo-classes (:first-of-type, :last-of-type) for custom styling?

Reply:
Yes, I believe it is a slight breaking change, because the pseudo-classes were more specific than the new classes, and therefore took higher precedence. I think https://github.com/mui/material-ui/pull/38520 also would've also been a slight breaking change in that sense?

Reply:
I believe merging this PR is worthwhile for resolving the associated issue (considering the upvotes in #12921), even if it involves a minor risk of breaking changes. Given our successful inclusion of #38520 in a previous patch release without any issues, I think we can do the same with this one.

My only concern is that developers who have copied this demo directly into their projects may be affected by the changes in this PR.

Let's keep this conversation thread open even after the PR gets merged.